packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -2641,6 +2605,1144 @@ test.serial('writeGranuleFromApi() removes preexisting granule file from postgre
+    fakeFileFactory({ size: 10 }),
+    fakeFileFactory({ size: 15 }),
+    fakeFileFactory({ size: 20 }),
+  ];
+
+  // set new createdAt and start time
+  const newCreatedAt = Date.now();
+  const partialUpdateGranule = {
+    ...completeGranule,
+    files: failUpdateFiles,
+    cmrLink: 'updatedGranuled.com',
+    status: 'running',
+    createdAt: newCreatedAt,
+  };
+
+  // create a new execution for the granule update operation
+  const { executionName, pgExecution, executionUrl } = await createGranuleExecution(t, 'running', t.context.stateMachineName);
+
+  cumulusMessage.cumulus_meta.workflow_start_time = newCreatedAt;
+  cumulusMessage.meta.status = 'running';
+  cumulusMessage.cumulus_meta.execution_name = executionName;
+  cumulusMessage.payload.granules[0] = partialUpdateGranule;
+
+  await writeGranulesFromMessage({
+    cumulusMessage,
+    executionCumulusId: pgExecution.cumulus_id,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  const dynamoRecord = await granuleModel.get({ granuleId });
+  const postgresRecord = await granulePgModel.get(knex, {
+    granule_id: granuleId,
+    collection_cumulus_id: collectionCumulusId,
+  });
+  const apiFormattedPostgresGranule
+    = await translatePostgresGranuleToApiGranule({
+      granulePgRecord: postgresRecord,
+      knexOrTransaction: knex,
+    });
+  const esRecord = await esGranulesClient.get(granuleId);
+
+  const expectedGranule = {
+    ...initialDynamoGranuleRecord,
+    // these original values should already be part of initialDynamoGranuleRecord as they were
+    // not updated but setting more explicitly for test transparency
+    files: files,
+    cmrLink: 'example.com',
+    productVolume: '15',
+    duration: initialDynamoGranuleRecord.duration,
+    // these are the only values that are updated in the granule
+    // some of the timestamp values are based on the operation
+    status: 'running',
+    createdAt: newCreatedAt,
+    timestamp: dynamoRecord.timestamp,
+    updatedAt: dynamoRecord.updatedAt,
+    // these values are updated due to the new execution
+    execution: executionUrl,
+  };
+
+  // Files array order is not promised to match between datastores
+  [esRecord, dynamoRecord, expectedGranule, apiFormattedPostgresGranule].forEach((record) => {
+    record.files.sort((f1, f2) => sortFilesByBuckets(f1, f2));
+  });
+
+  // Translated postgres granule matches expected updatedGranule
+  // minus model defaults
+  t.deepEqual(
+    apiFormattedPostgresGranule,
+    removeNilProperties(expectedGranule)
+  );
+  t.deepEqual(dynamoRecord, removeNilProperties(expectedGranule));
+  t.deepEqual(omit(esRecord, ['_id']), removeNilProperties(expectedGranule));
+});
+
+test.serial('writeGranulesFromMessage() on update changing granule status to "queued", with different files and a new execution, does not update the granule values or files in Postgres/DynamoDB/ES, so the pre-existing values and files will persist', async (t) => {
```

Comment:
wanted to flag this test as a particularly confusing one, see the comments right below the test description. with "queued" status and writeConstraints, the exclusion query that is built for executions is for any execution that exists in the DB.

Reply:
I'm not sure I follow. So if you have a granule in "queued" and a new execution compared to _any_ execution in the database, not just a different execution than what was previously on the granule?

Reply:
Discussed in a huddle. I had expected this test to behave the same as the above test with "running" status, where it did a limited granule update (status, time values) but not write files. But upon examining the results and the exclusion clause that is built, for "queued" status it seems if there are write constraints it will not update the granule if the execution exists at all in the db.